mutt/address.c

```diff
@@ -998,11 +997,17 @@ const char *mutt_addr_for_display(struct Address *a)
     return a->mailbox;
 
   local_mailbox = mutt_idna_intl_to_local(user, domain, MI_MAY_BE_IRREVERSIBLE);
-  if (!local_mailbox)
```

Comment:
I would prefer `user` and `domain` to be freed as soon as they're not needed anymore, no matter whether `mutt_idna_intl_to_local` succeeded or not. What you do in `mutt_addrlist_to_intl` is better in my opinion.

Reply:
Done!